pkg/collector/process_test.go

```diff
@@ -49,7 +49,7 @@ func TestTCPCollectingProcess_ReceiveTemplateRecord(t *testing.T) {
 		t.Fatalf("TCP Collecting Process does not start correctly: %v", err)
 	}
 	go func() {
-		time.Sleep(2 * time.Second)
+		time.Sleep(time.Millisecond)
```

Comment:
If we start the `go cp.Start()` before this go routine for fake exporter, do we still need the sleep?

Reply:
It is still possible to cause nil pointer problem if the collector is not ready. Changed to wait.Poll to wait for collector to be ready. Thanks!